### ISSUE TITLE
HTTP 500 code on invalid Host header

### ISSUE THREAD

**Issue by msilvacsw** [quarkusio/quarkus#39625]
### Describe the bug

Sending the following request to a quarkus 3.2.10.Final application:
`curl -v -X  'POST'  -H 'host: ZAP%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s' 'http://localhost:8070/endpoint'`

We get:
HTTP/1.1 500 Internal Server Error

The application gives a java.lang.NullPointerException. No Stack trace is printed, but debugging the issue we get:

```
        at io.vertx.core.net.impl.HostAndPortImpl.<init>(HostAndPortImpl.java:141)
        at io.vertx.core.net.HostAndPort.create(HostAndPort.java:20)
        at io.vertx.ext.web.impl.ForwardedParser.calculate(ForwardedParser.java:137)
        at io.vertx.ext.web.impl.ForwardedParser.absoluteURI(ForwardedParser.java:96)
        at io.vertx.ext.web.impl.HttpServerRequestWrapper.absoluteURI(HttpServerRequestWrapper.java:158)
        at io.quarkus.resteasy.runtime.standalone.VertxUtil.extractUriInfo(VertxUtil.java:31)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:107)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:93)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:576)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:840)

```

### Expected behavior

HTTP 400 error code

### Actual behavior

HTTP 500 error code

### How to Reproduce?

`curl -v -X  'POST'  -H 'host: ZAP%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s%n%s' 'http://localhost:8070/endpoint'`

### Output of `java -version`

openjdk version "17.0.10" 2024-01-16
OpenJDK Runtime Environment (build 17.0.10+7-Debian-1deb12u1)
OpenJDK 64-Bit Server VM (build 17.0.10+7-Debian-1deb12u1, mixed mode, sharing)

### Quarkus version or git rev

3.2.10.Final

**Comment by geoand** [comment]
Is this still a problem with Quarkus 3.8 or 3.9?

**Comment by geoand** [comment]
I am going to close this as it looks like a duplicate of #38230 which has been fixed in Quarkus since 3.7.

If this is not the case, please comment and we can look into it further
